services/QuillLMS/engines/evidence/lib/evidence/check/gen_ai.rb

```diff
@@ -38,23 +42,29 @@ def use_for_optimal_feedback? = true
 
       private def secondary_feedback_prompt = Evidence::GenAI::SecondaryFeedback::PromptBuilder.run(prompt:)
 
-      private def feedback_history
-        previous_feedback.map { |f| f['feedback'] }
+      private def history
+        previous_feedback.map do |feedback|
+          Evidence::GenAI::HistoryItem.new(
+            user: feedback[KEY_ENTRY]&.gsub(prompt.text, '')&.strip,
+            assistant: feedback[KEY_FEEDBACK]
+          )
+        end
       end
 
-      # TODO: This is a relative inefficient query. We'd likely want to update the Feedback#create API
-      # to save the user entry and feedback history (instead of just feedback history)
-      private def session_history
-        return [] if session.nil?
-
-        @history ||= session
-          .feedback_history
-          .sort_by(&:id)
-          .map { |fh| Evidence::GenAI::HistoryItem.new(user: fh.entry, assistant: fh.feedback_text) }
-      end
+      private def history_feedback_only = previous_feedback.map { |f| f[KEY_FEEDBACK] }
 
-      private def session
-        @session ||= Evidence.feedback_session_class.find_by(activity_session_uid: session_uid)
+      private def with_logging
```

Comment:
Is this logging temporary?
Do we want to restrict to staging environments?

Reply:
This is temporary, but I'd like to leave it in for a bit for debugging purposes on beta, but will leave a comment to that effect. 

This GenAI check should be off in production, so the additional `Rails.production?` check I'm planning to add there should, in effect, restrict this to non-prod.